### PR TITLE
iortcw: init at 1.51c

### PR DESCRIPTION
**Pull request by ghost**
###### Motivation for this change

Add engine for Return to Castle Wolfenstein game

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by gnidorah** [comment]
Tested with moddb version of rtcw, should be good to go

File: pkgs/games/iortcw/bare.nix [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+  NIX_CFLAGS_COMPILE = [
+    "-I${SDL2.dev}/include/SDL2"
+    "-I${opusfile}/include/opus"
+  ];
+  NIX_CFLAGS_LINK = [ "-lSDL2" ];
+
+  postInstall = ''
+    for i in `find $out/opt/iortcw -maxdepth 1 -type f -executable`
+    do
```
Comment (SuperSandro2000) [inline pkgs/games/iortcw/bare.nix:41]:
```suggestion
    for i in `find $out/opt/iortcw -maxdepth 1 -type f -executable`; do
```

File: pkgs/games/iortcw/default.nix [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  paths = [ sp mp ];
+
+  pathsToLink = [ "/opt" ];
+
+  buildInputs = [ makeWrapper ];
+
+  # so we can launch sp from mp game and vice versa
+  postBuild = ''
+    for i in `find -L $out/opt/iortcw -maxdepth 1 -type f -executable`
+    do
```
Comment (SuperSandro2000) [inline pkgs/games/iortcw/default.nix:20]:
```suggestion
    for i in `find -L $out/opt/iortcw -maxdepth 1 -type f -executable`; do
```

File: pkgs/top-level/all-packages.nix [hunk 3]
```diff
@@ -25084,6 +25084,11 @@ in
 
   instead-launcher = callPackage ../games/instead-launcher { };
 
+  # the package is splitted this way so iortcw forks could be easily built
+  # with iortcwBare.overrideAttrs and put into NUR
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:25088]:
```suggestion
  # the package is splitted this way so iortcw forks could be easily built with iortcwBare.overrideAttrs and put into NUR
```

**Comment by gnidorah** [comment]
@SuperSandro2000 Thanks for review!

File: pkgs/games/iortcw/bare.nix [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+    "USE_INTERNAL_LIBS=0"
+    "COPYDIR=${placeholder "out"}/opt/iortcw"
+    "USE_OPENAL_DLOPEN=0" "USE_CURL_DLOPEN=0"
+  ];
+
+  installTargets = [ "copyfiles" ];
+
+  buildInputs = [
+    opusfile libogg SDL2 freetype libjpeg openal
+    curl
```
Comment (SuperSandro2000) [inline pkgs/games/iortcw/bare.nix:29]:
```suggestion
    opusfile libogg SDL2 freetype libjpeg openal curl
```

File: pkgs/games/iortcw/bare.nix [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+  };
+
+  enableParallelBuilding = true;
+
+  sourceRoot = "source/SP";
+
+  makeFlags = [
+    "USE_INTERNAL_LIBS=0"
+    "COPYDIR=${placeholder "out"}/opt/iortcw"
+    "USE_OPENAL_DLOPEN=0" "USE_CURL_DLOPEN=0"
```
Comment (SuperSandro2000) [inline pkgs/games/iortcw/bare.nix:22]:
```suggestion
    "USE_OPENAL_DLOPEN=0"
    "USE_CURL_DLOPEN=0"
```

**Comment by gnidorah** [comment]
@SuperSandro2000 Thanks! 😃

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 100485` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>iortcw</li>
    <li>iortcwBare</li>
  </ul>
</details>

**Comment by gnidorah** [comment]
Calling some people commiting to pkgs/games to help with merge:
@abbradar @peterhoeg

**Comment by FRidh** [comment]
Please give the two packages a different description. What is `bare`?

**Comment by gnidorah** [comment]
@FRidh Let's try this way

**Comment by gnidorah** [comment]
@FRidh Thank you so much! :smile:

**Pull request merged**